### ISSUE TITLE
guiExample / ofParameter ofColor error :  no matching function for call to 'ofSetColor'

### ISSUE THREAD

**Issue by dimitre** [openframeworks/openFrameworks#7780]
```
examples/gui/guiExample/src/ofApp.cpp:56:2: error: no matching function for call to 'ofSetColor'
        ofSetColor(color);
        ^~~~~~~~~~
../../../libs/openFrameworks/graphics/ofGraphics.h:56:6: note: candidate function not viable: no known conversion from 'ofxColorSlider' (aka 'ofxColorSlider_<unsigned char>') to 'int' for 1st argument
void ofSetColor(int gray);
     ^
../../../libs/openFrameworks/graphics/ofGraphics.h:71:6: note: candidate template ignored: could not match 'ofColor_' against 'ofxColorSlider_'
void ofSetColor(const ofColor_<T> & acolor) {
     ^
../../../libs/openFrameworks/graphics/ofGraphics.h:57:6: note: candidate function not viable: requires 2 arguments, but 1 was provided
void ofSetColor(const ofColor & acolor, int a);
     ^
../../../libs/openFrameworks/graphics/ofGraphics.h:39:6: note: candidate function not viable: requires 3 arguments, but 1 was provided
void ofSetColor(int r, int g, int b);
     ^
../../../libs/openFrameworks/graphics/ofGraphics.h:55:6: note: candidate function not viable: requires 4 arguments, but 1 was provided
void ofSetColor(int r, int g, int b, int a);
     ^

```

**Comment by artificiel** [comment]
it seems this template (seems recent, with the #7650) fails to be "deduced" as a candidate for the ofColor operator of ofxColorSlider because it tries to process the ofxColorSlider itself:

https://github.com/openframeworks/openFrameworks/blob/dd09e975ad1166ea0cb48edeae35c1f75dc9ed4b/libs/openFrameworks/graphics/ofGraphics.h#L70-L74

https://github.com/openframeworks/openFrameworks/blob/dd09e975ad1166ea0cb48edeae35c1f75dc9ed4b/addons/ofxGui/src/ofxSliderGroup.h#L52

adding an explicit `ofSetColor(ofColor aColor);` method (one without the alpha param) fixes the specific case of the reported bug, but not sure how to properly write things for the templated ofxColorSlider to follow the templated ofColor as it is presumed it should?

**Comment by NickHardeman** [comment]
Hmm, that's a tricky one. I think the conversion constructors in ofColor might also cause some further confusion with ofSetColor using a ColorSlider, ie ofxFloatColorSlider passed to ofSetColor( ofColor color ) might be converted to ofColor, when ofFloatColor is desired. 
I wonder if removing the ofSetColor templated function and adding explicit ones would solve the issue.
```
ofSetColor( const ofColor& color);
ofSetColor( const ofFloatColor& color);
ofSetColor( const ofShortColor& color);
```

**Comment by artificiel** [comment]
it seems like the simple solution as anyway there are only 3 template cases so not much duplication, and it's not like users are creating their own ofColor types? (for sure ofxColorSlider itself is hard-coded to the 3 types).

**Comment by NickHardeman** [comment]
Closed by #7788
